File: codeingame/thor/thor.lua
-- lightX: the X position of the light of power
-- lightY: the Y position of the light of power
-- initialTX: Thor's starting X position
-- initialTY: Thor's starting Y position
next_token = string.gmatch(io.read(), "[^%s]+")
lightX = tonumber(next_token())
lightY = tonumber(next_token())
initialTX = tonumber(next_token())
initialTY = tonumber(next_token())

x = initialTX;
y = initialTY;

-- game loop
while true do

      dx = lightX - x;
      dy = y - lightY;
      dir = "";

      if (dy > 0) then
         dir  = "N"
         y = y - 1
      end
      if (dy < 0) then
        dir  = "S"
        y = y + 1;
      end
      if (dx > 0) then
        dir = dir .. "E"
        x = x + 1;
      end
      if (dx < 0) then
        dir = dir .. "W";
        x = x - 1;
      end

    print( dir )
end

File: codeingame/thor/thor.js
/**
 * Thor problem on codingame.com
 *
 * 2015, dwhitnee@gmail.com
 */

var inputs = readline().split(' ');
var LX = parseInt(inputs[0]); // the X position of the light of power
var LY = parseInt(inputs[1]); // the Y position of the light of power
var TX = parseInt(inputs[2]); // Thor's starting X position
var TY = parseInt(inputs[3]); // Thor's starting Y position

var x = TX;
var y = TY;

// game loop
while (true) {
  var dx = LX - x;
  var dy = y - LY;
  var dir = "";

  if (dy > 0) {  dir  = "N"; y -= 1; }
  if (dy < 0) {  dir  = "S"; y += 1; }
  if (dx > 0) {  dir += "E"; x += 1; }
  if (dx < 0) {  dir += "W"; x -= 1; }
  print( dir );
}
